src/Foundation/Workflow/code/Actions/SubmissionNotification.cs

```diff
@@ -10,124 +10,129 @@
+			Error.Assert(mailServer.Length > 0, "The 'Mail server' field is not specified in the mail action item: " + actionPath);
+
+			message = ReplaceVariables(message, args);
+
+			MailMessage mailMessage = new MailMessage
+			{
+				From = new MailAddress(fromAddress),
+				Subject = subject,
+				Body = message,
+				IsBodyHtml = true
+			};
+
+			AddRecipientsToMail(mailMessage, approversRole);
+
+			try
+			{
+				SmtpClient smtpClient = new SmtpClient(mailServer);
+				smtpClient.Send(mailMessage);
+			}
+			catch (Exception ex)
+			{
+				Log.Error($"Failed to send mail: {ex.Message}", ex, typeof(SubmissionNotification));
+			}
+		}
+
+		/// <summary>
+		/// Gets the text.
+		/// </summary>
+		/// <param name="commandItem">The command item.</param>
+		/// <param name="field">The field.</param>
+		/// <param name="args">The arguments.</param>
+		/// <returns></returns>
+		private string GetText(Item commandItem, string field, WorkflowPipelineArgs args)
+		{
+			string text = commandItem[field];
+			if (text.Length > 0)
+			{
+				return this.ReplaceVariables(text, args);
+			}
+			return string.Empty;
+		}
+
+		/// <summary>
+		/// Replaces the variables.
+		/// </summary>
+		/// <param name="text">The text.</param>
+		/// <param name="args">The arguments.</param>
+		/// <returns></returns>
+		private string ReplaceVariables(string text, WorkflowPipelineArgs args)
+		{
+			Item contentItem = args.DataItem;
+
+			var contentWorkflow = contentItem.Database.WorkflowProvider.GetWorkflow(contentItem);
+			var contentHistory = contentWorkflow.GetHistory(contentItem);
+
+			if (contentHistory.Length > 0)
+			{
+				var lastUser = contentHistory.Last().User;
+				text = text.Replace("$submitter$", lastUser);
+			}
+
+			text = text.Replace("$itemPath$", args.DataItem.Paths.FullPath);
+			text = text.Replace("$itemLanguage$", args.DataItem.Language.ToString());
+			text = text.Replace("$itemVersion$", args.DataItem.Version.ToString());
+			text = text.Replace("$hostname$", HttpContext.Current.Request.Url.Host);
+			text = text.Replace("$comment$", args.CommentFields["Comments"]);
+			text = text.Replace("$actionedby$", User.Current.Name);
+
+			return text;
+		}
+
+		public static MailMessage AddRecipientsToMail(MailMessage mailMessage, string roleName)
+		{
+			if (Role.Exists(roleName))
+			{
+				Role role = Role.FromName(roleName);
+				List<User> users = RolesInRolesManager.GetUsersInRole(role, true)
+					.Where(x => x.IsInRole(role))
+					.Where(user => !string.IsNullOrEmpty(user.Profile.Email))
+					.ToList();
+
+				foreach (User user in users)
+				{
+					mailMessage.To.Add(user.Profile.Email);
+				}
+			}
+			else if (User.Exists(roleName))
+			{
+				User user = User.FromName(roleName, false);
+				mailMessage.To.Add(user.Profile.Email);
+			}
+			else
+			{
+				Log.Error($"No Users with valid email addresses found in role {roleName}", typeof(SubmissionNotification));
```

Comment:
As we change `roleName` to `principalName`, we should update this message to `No users with valid email addresses found for security principal {principalName}`

Reply:
sure will do it.

Reply:
Thanks for all your valuable feedback